Src/Json/Shared.Json/ObjectDiffPatch.cs

```diff
@@ -0,0 +1,378 @@
+﻿#region *   License     *
+/*
+    SimpleHelpers - ObjectDiffPatch   
```

Comment:
I did a quick test, plainly using

            var subject = JToken.Parse("{child:{subject:'foo'}}");
            var expected = JToken.Parse("{child:{expected:'bar'}}");
            subject.ShouldBeEquivalentTo(expected);

which unexpectedly did not fail. 

The equivalency steps handling the JTokens were

`CanHandle == true`:

- `TryConversionEquivalencyStep`
- `ReferenceEqualityEquivalencyStep`
- `GenericEnumerableEquivalencyStep`
- `EnumerableEquivalencyStep`
- `StructuralEqualityEquivalencyStep`

`Handle == true`:

- `ValueTypeEquivalencyStep` (not sure where this one came from)
- `StructuralEqualityEquivalencyStep`

`Any/FirstOrDefault( CanHandle && Handle)`:

- GenericEnumerableEquivalencyStep

Right now i want to stop digging on the graph equivalency API as it seems too complicated for me, especially the expression based `GenericEnumerableEquivalencyStep` and the coupling between steps, validator & context.